dist/index.js

```diff
@@ -55343,24 +55346,30 @@ async function getRegistrationToken() {
   }
 }
 
+// remove all runners with the input label
 async function removeRunner() {
-  const runner = await getRunner(config.input.label);
   const octokit = github.getOctokit(config.input.githubToken);
 
-  // skip the runner removal process if the runner is not found
-  if (!runner) {
-    core.info(`GitHub self-hosted runner with label ${config.input.label} is not found, so the removal is skipped`);
+  // find all runners with our label
+  const allRunners = await octokit.paginate('GET /repos/{owner}/{repo}/actions/runners', config.githubContext);
+  const runners = _.filter(allRunners, { labels: [{ name: config.input.label }] });
+
+  // skip the runner removal process if the runners are not found
+  if (runners.length < 1) {
+    core.info(`GitHub self-hosted runner(s) with label ${config.input.label} not found, so the removal is skipped`);
     return;
   }
 
-  try {
-    await octokit.request('DELETE /repos/{owner}/{repo}/actions/runners/{runner_id}', _.merge(config.githubContext, { runner_id: runner.id }));
-    core.info(`GitHub self-hosted runner ${runner.name} is removed`);
-    return;
-  } catch (error) {
-    core.error('GitHub self-hosted runner removal error');
-    throw error;
+  for (let runner of runners) {
+    try {
+      await octokit.request('DELETE /repos/{owner}/{repo}/actions/runners/{runner_id}', _.merge(config.githubContext, { runner_id: runner.id }));
```

Comment:
This looks brittle, we don't rety and were making calls over the network, we could easily leave runners running because of transient blips. 

I suspect we will need to revisit this anyway as we will be moving to ephemeral runners, maybe we should look at moving to ephemeral runners sooner as it means we don't have to invest time in the shutdown code we will be redoing again later ?

Reply:
I agree this is could be brittle. However, this is just a loop over the original code, so not that much more brittle than stock ;-) Risk is also much lower, if this call fails we just get left with [a runner entry on GitHub](https://github.com/fac/freeagent/settings/actions/runners) in disconnected state. No cost as far as I know. It is the AWS calls that actually remove the instances.

It should probably try and remove them all and not throw on the first error.

However, as you note, I think ephemeral runners is they real way to go, then we wont even need to run this code, they will self de-register. So far I have not actually seen it fail, so tempted to merge as is and patch if we get problems and it is needed or move to ephemeral quicker.